### DISCUSSION TITLE
Horizontal Alignment of left spines with linked x axis

### DISCUSSION THREAD

**Discussion by zariiii9003** [pyqtgraph/pyqtgraph#2282] [General]
I have three `PlotItem` in a single `GraphicsLayoutWidget`. The x-axis of the second and third `PlotItem` is linked to the first.
How can i align the left spines of the three plots so they can share a single x-axis?

![image](https://user-images.githubusercontent.com/52598363/165920207-2cf426eb-42f7-495f-8b67-6a581a75ebec.png)

**Comment by ixjlyons** [comment]
You can set each "left" `AxisItem` to have a fixed width, like `p1.getAxis("left").setWidth(60)`. This disables the auto-expand/shrink behavior.

```python
import numpy as np
import pyqtgraph as pg

app = pg.mkQApp()

win = pg.GraphicsLayoutWidget()

x = np.linspace(0, 1, 100)
y = np.sin(2 * np.pi * 4 * x) + 1.0

p1 = win.addPlot(x=x, y=50000*y)
win.nextRow()
p2 = win.addPlot(x=x, y=100*y)
win.nextRow()
p3 = win.addPlot(x=x, y=10*y)

p2.setXLink(p1)
p3.setXLink(p1)

p1.getAxis("left").setWidth(60)
p2.getAxis("left").setWidth(60)
p3.getAxis("left").setWidth(60)

win.show()

app.exec()
```

If you still need adaptive width, you'd have to manage that yourself. See `AxisItem._updateWidth`.

**Reply by zariiii9003** [reply]
That works, thank you very much! I actually tried `setFixedWidth()` before and it didn't do anything but i guess i was on the right track
